lib/core/src/Cardano/Wallet/Primitive/AddressDerivation/Random.hs

```diff
@@ -254,6 +267,19 @@ decodeAddressDerivationPath rootKey = do
         fail $ "decodeAddressDerivationPath: type is not 0 (public key), it is " ++ show addrType
     pure path
 
+decodeAddressDerivationPathShort
+    :: Key 'RootK XPub
+    -> CBOR.Decoder s (Maybe (Index 'Hardened 'AccountK, Index 'Soft 'AddressK))
+decodeAddressDerivationPathShort rootKey = do
+    _ <- CBOR.decodeListLenCanonicalOf 3
+    _ <- CBOR.decodeBytes -- Address Root
+    len <- CBOR.decodeMapLen -- Address Attributes
+    case len of
+        1 -> do
+            _ <- CBOR.decodeWord8 -- Type (1)
+            decodeDerivationPath rootKey
+        _ -> fail $ "decodeAddressDerivationPath: Incorrect map length: " ++ show len
```

Comment:
Not sure what's the rationale behind this function :thinking: ? 
Note that `decodeAddressDerivationPath` as it stands right now is necessary to handle both mainnet and testnet addresses (as there are several attributes in the case of testnet).

Reply:
this function will be erased when I correct `toAddress` . toAddress works with this one and property test passes